canvas_sdk/commands/base.py

```diff
@@ -14,10 +13,9 @@ class _BaseCommand(BaseModel):
     class Meta:
         key = ""
 
-    # todo: update int to str as we should use external identifiers
-    note_id: int | None = None
+    note_id: str | None = None
     command_uuid: str | None = None
-    user_id: int
+    user_id: str
 
```

Comment:
Looks like the `auth_user` table does not contain a UUID value, only `id` and `username`. Should we add one?

Reply:
sorry i dropped the ball on this the last week or so, hoping to close it out today/tomorrow. 

i'm thinking the `staff_key` will actually make sense to use here, since we'd only expect commands to be acted upon by a staff user. thoughts @jrwils @aduane ?

Reply:
I think we have some examples where a questionnaire is originated by a patient via API, so to allow for that we would need to reference a user rather than patient or staff. (Both patients and staff have user records)

Should we:
1. add a uuid to the users table
2. restrict command utilization in plugins to staff
3. have staff_key and patient_key as separate attributes, and one and only one must be used

(2) seems like the least work, but what if an event fires on a command originated by a patient?


Regardless, I would suggest we de-prioritize this, as command manipulation is not part of our initial launch feature set.